### PR TITLE
Resolve incorrectly-set test-version-utils config

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

#16108 converted test-version-utils to use esnext modules, which accidentally made import of `compatOptions.js` asynchronous. Mocha doesn't really support this (they cannot load ESM modules as part of configuration--see documented limitation in code--and there's no way to run async module-loading code of our own as part of mocharc).
This ended up resulting in us not reporting test results with all the context that they should have (both in CI pipeline and in telemetry contexts).

For now, I've converted the nconf option loading in `compatOptions.ts` to be written in CJS so that it's accessible to both mocharc and at test run time.
To minimize the amount of code this impacts, I also reorganized the setup of `baseVersion` into a different file.

[AB#4872](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4872)

### CODE REVIEW

**Comment by Abe27342** [comment]
reviewer can verify this works by comparing the number of tests reported in the PR build to similar runs (this one is ~3k more--see [here](https://dev.azure.com/fluidframework/public/_build/results?buildId=172715&view=ms.vss-test-web.build-test-results-tab) and diff another successful run)

File: packages/test/test-version-utils/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
 	"sideEffects": false,
 	"type": "module",
 	"main": "dist/index.js",
-	"module": "lib/index.js",
+	"module": "dist/index.js",
```
Comment (Abe27342) [inline packages/test/test-version-utils/package.json:16]: noticed I never correctly updated this with the change to esnext modules.

File: packages/test/test-version-utils/mocharc-common.cjs [hunk 2]
```diff
@@ -4,7 +4,7 @@
  */
 
 "use strict";
-const options = import("./dist/compatOptions.js");
```
Comment (Abe27342) [inline packages/test/test-version-utils/mocharc-common.cjs:7]: This is the actually problematic line: `import` returns a promise to the contents of the module. So the subsequent logic in `getFluidTestVariant` wouldn't work.

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM, out of curiosity, is there a task to convert this to .ts file?

**Comment by Abe27342** [comment]
> LGTM, out of curiosity, is there a task to convert this to .ts file?

I'm not convinced mocha will add support for this on on any soon-ish timeframe (if ever, it doesn't see much active development because the set of consumers it has to support is quite large). I could see us using a different testing framework before mocha gave better ecosystem support for esm honestly, despite that being the way the ecosystem is heading.

Writing this as a TS file is doable and I considered it, but it adds a lot more complexity to the build setup of this package (moreso than just using js in my opinion) without significant value: the interesting stuff in that file is all relating to `nconf` and probably won't be touched much. So I'd rather keep things simpler to understand rather than prioritize benefits one gets from a type system.

**Pull request merged**